network/server/tchannelthrift/node/service.go

```diff
@@ -332,9 +341,16 @@ func (s *service) FetchBlocksMetadataRaw(tctx thrift.Context, req *rpc.FetchBloc
 		includeChecksums = *req.IncludeChecksums
 	}
 
-	fetched, nextPageToken, err := s.db.FetchBlocksMetadata(
-		ctx, s.idPool.GetBinaryID(ctx, req.NameSpace), uint32(req.Shard),
-		start, end, req.Limit, pageToken, includeSizes, includeChecksums)
+	buffer := s.bytesPool.Get(len(req.NameSpace))
```

Comment:
What's the point in pooling these bytes? `req.NameSpace` is allocated by thrift outside of pools, why copy it into a pooled memory?